src/scrape_up/github/pull_request.py

```diff
@@ -131,3 +131,173 @@ def reviewers(self):
+                "data": merged_at,
+                "message": f"Found merge date and time for {self.pr_number}",
+            }
+        except:
+            message = f"No merge date and time found for {self.pr_number}"
+            return {
+                "data": None,
+                "message": message,
+            }
+
+    def merged_by(self):
+        """
+        Fetch the name of the user who merged the pull request if it has been merged
+        """
+        data = self.__scrape_page()
+        try:
+            merged_by_element = data.find("a", class_="commit-author")
+            merged_by = merged_by_element.text.strip()
+            return {
+                "data": merged_by,
+                "message": f"Found merge author for {self.pr_number}",
+            }
+        except:
+            message = f"No merge author found for {self.pr_number}"
+            return {
+                "data": None,
+                "message": message,
+            }
+
+    def comments_count(self):
+        """
+        Fetch the number of comments on a pull request
+        """
+        data = self.__scrape_page()
+        try:
+            comments_count_element = data.find("a", class_="js-selected-navigation-item subnav-item")
+            comments_count = comments_count_element.find("span", class_="Counter").text.strip()
+            return {
+                "data": comments_count,
+                "message": f"Found {comments_count} comments for {self.pr_number}",
+            }
+        except:
+            message = f"No comments found for {self.pr_number}"
+            return {
+                "data": None,
+                "message": message,
+            }
+
+    def review_comments_count(self):
+        """
+        Fetch the number of review comments on a pull request
+        """
+        data = self.__scrape_page()
+        try:
+            review_comments_count_element = data.find("a", class_="js-selected-navigation-item subnav-item")
+            review_comments_count = review_comments_count_element.find_all("span", class_="Counter")[1].text.strip()
+            return {
+                "data": review_comments_count,
+                "message": f"Found {review_comments_count} review comments for {self.pr_number}",
+            }
+        except:
+            message = f"No review comments found for {self.pr_number}"
+            return {
+                "data": None,
+                "message": message,
+            }
+
+    def commits_url(self):
+    """
+    Get the URL to fetch the list of commits in a pull request
+    """
+    return f"https://github.com/{self.username}/{self.repository}/pull/{self.pr_number}/commits" if self.pr_number else None
+
+    def diff_url(self):
+    """
+    Get the URL to view the diff of changes in a pull request
+    """
+    return f"https://github.com/{self.username}/{self.repository}/pull/{self.pr_number}.diff" if self.pr_number else None
+
+    def html_url(self):
+    """
```

Comment:
Hey @shivvamm, please follow the code structure of the project while returning the output as shown in the picture below.
![image](https://github.com/Clueless-Community/scrape-up/assets/93156825/eed0414e-4da2-4c0b-afe0-ffbdf225c2ca)

Reply:
Okay sure I'll update it now

Reply:
Done I fixed it